mathjax3-ts/output/chtml/Wrappers/maction.ts

```diff
@@ -78,38 +82,133 @@ export class CHTMLmaction<N, T, D> extends CHTMLWrapper<N, T, D> {
      * The valid action types and their handlers
      */
     public static Actions = new Map([
-        ['toggle', node => {
+        ['toggle', [(node, data) => {
+            //
+            // Mark which child is selected
+            //
             node.adaptor.setAttribute(node.chtml, 'toggle', node.node.attributes.get('selection') as string);
-        }],
+            //
+            // Cache the data needed to select another node
+            //
+            const math = node.factory.chtml.math;
+            const document = node.factory.chtml.document;
+            const mml = node.node as MmlMaction;
+            //
+            // Add a click handler that changes the selection and rerenders the expression
+            //
+            node.setEventHandler('click', (event: Event) => {
+                if (!math.start.node) {
+                    //
+                    // If the MathItem was created by hand, it might not have a node
+                    // telling it where to replace the existing math, so set it.
```

Comment:
I don't understand that.

Reply:
We discussed this face-to-face, but just or the record, there is a common use case where you create a `MathItem` by hand (for things like converting a TeX string to an HTML string), and that `MathItem` is not attached to a document directly.  This is done in the TeX and MathML labs, for instance, where the TeX or MathML is converted without it being part of a document, and the resulting HTML node is inserted into the page.  

This takes care of setting the node in those cases, so that a toggle action can remove the old HTML and replace it by the newly re-rendered HTML.